stress/src/async_throughput.rs

```diff
@@ -0,0 +1,259 @@
+        println!(
+            "{}: {} ops ({} successful, {} failed) in {:.2}s = {:.2} ops/sec",
+            label,
+            self.completed_ops,
+            self.successful_ops,
+            self.failed_ops,
+            self.duration.as_secs_f64(),
+            self.throughput
+        );
+    }
+}
+
+/// Configuration for the throughput test
+pub struct ThroughputConfig {
+    /// Number of concurrent operations
+    pub concurrency: usize,
+    /// Reporting interval for continuous tests
+    pub report_interval: Duration,
+    /// Optional target number of operations (None for continuous)
+    pub target_ops: Option<usize>,
+}
+
+impl Default for ThroughputConfig {
+    fn default() -> Self {
+        Self {
+            concurrency: 100,
+            report_interval: Duration::from_secs(5),
+            target_ops: None,
+        }
+    }
+}
+
+/// Generic stream-based throughput tester
+pub struct ThroughputTest;
+
+impl ThroughputTest {
+    /// Run a continuous throughput test (until interrupted)
+    pub async fn run_continuous<F, Fut, T, E>(
+        name: &str,
+        config: ThroughputConfig,
+        operation_factory: F,
+    ) -> ThroughputStats
+    where
+        F: Fn() -> Fut + Send + Sync + 'static,
+        Fut: Future<Output = Result<T, E>> + Send + 'static,
+        T: Send + 'static,
+        E: std::fmt::Display + Send + 'static,
+    {
+        println!(
+            "Testing {} with concurrency level: {}",
+            name, config.concurrency
+        );
+
+        let completed_ops = Arc::new(AtomicU64::new(0));
+        let successful_ops = Arc::new(AtomicU64::new(0));
+        let failed_ops = Arc::new(AtomicU64::new(0));
+
+        // Clone for reporting thread
+        let completed_clone = Arc::clone(&completed_ops);
+        let successful_clone = Arc::clone(&successful_ops);
+        let failed_clone = Arc::clone(&failed_ops);
+        let start_time = Instant::now();
+        let report_interval = config.report_interval;
+
+        // Spawn reporting thread
+        let _reporting_handle = std::thread::spawn(move || loop {
+            std::thread::sleep(report_interval);
+            let ops = completed_clone.load(Ordering::Relaxed);
+            let success = successful_clone.load(Ordering::Relaxed);
+            let failed = failed_clone.load(Ordering::Relaxed);
+            let elapsed = start_time.elapsed();
+            let throughput = ops as f64 / elapsed.as_secs_f64();
+
+            println!(
+                "Progress: {} ops completed ({} successful, {} failed) in {:.2}s = {:.2} ops/sec",
+                ops,
+                success,
+                failed,
+                elapsed.as_secs_f64(),
+                throughput
+            );
+        });
+
+        // Create operation factory Arc for sharing
+        let operation_factory = Arc::new(operation_factory);
+
+        // Create infinite stream of operations
+        let mut operation_stream = stream::iter(0..)
```

Comment:
We need to spawn tasks for these operations to utilize the worker threads:

```rust
let mut operation_stream = stream::iter(0..)
    .map(|_| tokio::task::spawn(operation_factory()))
    .buffer_unordered(config.concurrency);
```

Reply:
Thanks for fixing this, done.